WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginWpcomService.java

```diff
@@ -0,0 +1,384 @@
+    private static final String ARG_SOCIAL_LOGIN = "ARG_SOCIAL_LOGIN";
+    private static final String ARG_SOCIAL_SERVICE = "ARG_SOCIAL_SERVICE";
+
+    public enum LoginPhase {
+        IDLE,
+        AUTHENTICATING(25),
+        SOCIAL_LOGIN(25),
+        FETCHING_ACCOUNT(50),
+        FETCHING_SETTINGS(75),
+        FETCHING_SITES(100),
+        SUCCESS,
+        FAILURE_EMAIL_WRONG_PASSWORD,
+        FAILURE_2FA,
+        FAILURE_SOCIAL_2FA,
+        FAILURE_FETCHING_ACCOUNT,
+        FAILURE_CANNOT_ADD_DUPLICATE_SITE,
+        FAILURE;
+
+        public final int progressPercent;
+
+        LoginPhase() {
+            this.progressPercent = 0;
+        }
+
+        LoginPhase(int progressPercent) {
+            this.progressPercent = progressPercent;
+        }
+
+        public boolean isInProgress() {
+            return this != LoginPhase.IDLE && !isTerminal();
+        }
+
+        public boolean isError() {
+            return this == LoginPhase.FAILURE
+                    || this == LoginPhase.FAILURE_EMAIL_WRONG_PASSWORD
+                    || this == LoginPhase.FAILURE_2FA
+                    || this == LoginPhase.FAILURE_SOCIAL_2FA
+                    || this == LoginPhase.FAILURE_FETCHING_ACCOUNT
+                    || this == LoginPhase.FAILURE_CANNOT_ADD_DUPLICATE_SITE;
+        }
+
+        public boolean isTerminal() {
+            return this == LoginPhase.SUCCESS || isError();
+        }
+    }
+
+    public static class OnLoginStateUpdated {
+        public final LoginPhase state;
+
+        OnLoginStateUpdated(LoginPhase state) {
+            this.state = state;
+        }
+    }
+
+    static class OnCredentialsOK {
+        OnCredentialsOK() {}
+    }
+
+    @Inject Dispatcher mDispatcher;
+    @Inject AccountStore mAccountStore;
+    @Inject SiteStore mSiteStore;
+
+    private LoginPhase mLoginPhase = LoginPhase.IDLE;
+
+    private String mIdToken;
+    private String mService;
+    private boolean isSocialLogin;
+
+    public static void loginWithEmailAndPassword(
+            Context context,
+            String email,
+            String password,
+            String idToken, String service,
+            boolean isSocialLogin) {
+        Intent intent = new Intent(context, LoginWpcomService.class);
+        intent.putExtra(ARG_EMAIL, email);
+        intent.putExtra(ARG_PASSWORD, password);
+        intent.putExtra(ARG_SOCIAL_ID_TOKEN, idToken);
+        intent.putExtra(ARG_SOCIAL_SERVICE, service);
+        intent.putExtra(ARG_SOCIAL_LOGIN, isSocialLogin);
+        context.startService(intent);
+    }
+
+    public static void clearLoginServiceState() {
+        OnLoginStateUpdated onLoginStateUpdated = EventBus.getDefault().removeStickyEvent(OnLoginStateUpdated.class);
+        if (onLoginStateUpdated != null && onLoginStateUpdated.state.isTerminal()) {
+            EventBus.getDefault().removeStickyEvent(OnLoginStateUpdated.class);
```

Comment:
I'm not sure why `removeStickyEvent` is called twice on the `OnLoginStateUpdated.class` here - I believe in some other part of the code there's the assumption that 2 sticky events are posted when the last sticky one is `isTerminal()`?  but I'm only guessing here. If we need to call `removeStickyEvent` twice let's add a comment to clarify what's happening here (i.e. what the first removal does, and what the second one does, and when that's needed)

Reply:
Good catch! This was actually a bad merge situation where extra code was left in. Sorry for the confusion! Sticky event just needs to be removed once. Fixed with 3277502.